### PR TITLE
Vue: Consistent slot, support Vue3

### PR DESCRIPTION
**Pull request by fisker**
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-html/printer-html.js [hunk 1]
```diff
@@ -1102,7 +1110,7 @@ function printEmbeddedAttributeValue(node, originalTextToDoc, options) {
     /**
      *     v-if="jsExpression"
      */
-    const jsExpressionBindingPatterns = ["^v-", "^#"];
+    const jsExpressionBindingPatterns = ["^v-"];
```
Comment (fisker) [inline src/language-html/printer-html.js:1113]: The `#default="..."` is added in #9609, and it should not be here.

Comment (wenfangdu) [reply src/language-html/printer-html.js:1113]: Ok, thanks for mentioning.

File: src/language-html/printer-html.js [hunk 2]
```diff
@@ -1082,9 +1083,16 @@ function printEmbeddedAttributeValue(node, originalTextToDoc, options) {
...
       return printVueFor(getValue(), textToDoc);
     }
-
-    if (node.fullName === "slot-scope") {
-      return printVueSlotScope(getValue(), textToDoc);
+    if (
+      node.fullName.charAt(0) === "#" ||
+      node.fullName === "slot-scope" ||
+      node.fullName === "v-slot" ||
+      node.fullName.startsWith("v-slot:") ||
```
Comment (sosukesuzuki) [inline src/language-html/printer-html.js:1090]: What do you think about to extract as function named `isSlotNode` and `isSfcSyntaxSugarNode`  to improve readability? like below:

```js
function isSlotNode(node) {
  return (
    node.fullName.charAt(0) === "#" ||
    node.fullName === "slot-scope" ||
    node.fullName === "v-slot" ||
    node.fullName.startsWith("v-slot:")
  );
}
```
```js
// https://github.com/vuejs/rfcs/blob/sfc-improvements/active-rfcs/0000-sfc-script-setup.md
// https://github.com/vuejs/rfcs/blob/sfc-improvements/active-rfcs/0000-sfc-style-variables.md
function isSfcSyntaxSugarNode(node, options) {
  return (
    isVueSfcBlock(node.parent, options) &&
    ((node.parent.fullName === "script" && node.fullName === "setup") ||
      (node.parent.fullName === "style" && node.fullName === "vars"))
  );
}
```

Comment (fisker) [reply src/language-html/printer-html.js]: Done

**Review by sosukesuzuki — APPROVED** [review]

**Pull request merged**